File: Java/SelectionSort.java

public class SelectionSort {
	
	public static void main(String [] args){
		
		int vetor[] = {9, 0, 4, 2, 3, 8, 7, 1, 6, 5};

		System.out.println("Selection Sort:");
		System.out.println("Vetor não ordenado:");
		mostraVetor(vetor);
		
		vetor = selectionSort(vetor, 0);
		
		System.out.println("Vetor ordenado:");
		mostraVetor(vetor);
	}
	
	public static int[] selectionSort(int vetor[], int indice){
		if(indice >= vetor.length-1)
			return vetor;
		
		int minIndice = indice;
		
		for (int i = indice+1; i < vetor.length; i++)
			if(vetor[i] < vetor[minIndice])
				minIndice = i;
		
		int temp = vetor[indice];
		vetor[indice] = vetor[minIndice];
		vetor[minIndice] = temp;
		
		selectionSort(vetor, indice+1);
		
		return vetor;
	}
	
	public static void mostraVetor(int vetor[]){
		for (int i = 0; i < vetor.length; i++){
			System.out.print(vetor[i] + ", ");
		}
		System.out.println("");
	}
}

File: src/python/selection_sort.py
"""Implementation of a selection sort algorithm with recursion."""

def selection_sort(arr, index):
    """
    Implementation of a selection sort algorithm with recursion.
    Args:
    arr (list): The list to be sorted.
    index (int): The index of the element to be sorted in the list.
    """
    if index >= len(arr) - 1:
        return

    # min_index holds the position of the smallest value relative to the index
    min_index = index

    for i in range(index + 1, len(arr)):
        if arr[i] < arr[min_index]:
            min_index = i

    temp = arr[index]
    arr[index] = arr[min_index]
    arr[min_index] = temp

    selection_sort(arr, index + 1)

unsorted_list = [82, 83, 92, 12, 23, 45, 64, 91, 73]

print(unsorted_list)
selection_sort(unsorted_list, 0)
print(unsorted_list)


